spec/models/customstates.rb

```diff
@@ -1,52 +1,47 @@
 # -*- encoding : utf-8 -*-
-module InfoRequestCustomStates
-
-  def self.included(base)
-    base.extend(ClassMethods)
-  end
+class InfoRequestThemeStates
 
   # Mixin methods for InfoRequest
-  def theme_calculate_status
-    return 'waiting_classification' if self.awaiting_description
-    waiting_response = self.described_state == "waiting_response" || self.described_state == "deadline_extended"
-    return self.described_state unless waiting_response
-    if self.described_state == 'deadline_extended'
+  def calculate_status(info_request)
+    return 'waiting_classification' if info_request.awaiting_description
+    waiting_response = info_request.described_state == "waiting_response" || info_request.described_state == "deadline_extended"
+    return info_request.described_state unless waiting_response
+    if info_request.described_state == 'deadline_extended'
       return 'deadline_extended' if
-      Time.now.strftime("%Y-%m-%d") < self.date_deadline_extended.strftime("%Y-%m-%d")
+      Time.now.strftime("%Y-%m-%d") < date_deadline_extended(info_request).strftime("%Y-%m-%d")
       return 'waiting_response_very_overdue'  if
-      Time.now.strftime("%Y-%m-%d") > Holiday.due_date_from_working_days(self.date_deadline_extended, 15).strftime("%Y-%m-%d")
+      Time.now.strftime("%Y-%m-%d") > Holiday.due_date_from_working_days(date_deadline_extended(info_request), 15).strftime("%Y-%m-%d")
       return 'waiting_response_overdue'
     end
     return 'waiting_response_very_overdue' if
-    Time.now.strftime("%Y-%m-%d") > self.date_very_overdue_after.strftime("%Y-%m-%d")
+    Time.now.strftime("%Y-%m-%d") > info_request.date_very_overdue_after.strftime("%Y-%m-%d")
     return 'waiting_response_overdue' if
-    Time.now.strftime("%Y-%m-%d") > self.date_response_required_by.strftime("%Y-%m-%d")
+    Time.now.strftime("%Y-%m-%d") > info_request.date_response_required_by.strftime("%Y-%m-%d")
     return 'waiting_response'
   end
 
-  def date_deadline_extended
+  def date_deadline_extended(info_request)
     # TODO: shouldn't this be 15 days after the date the status was
     # changed to "deadline extended"? Or perhaps 15 days ater the
     # initial request due date?
-    return Holiday.due_date_from_working_days(self.date_response_required_by, 15)
+    return Holiday.due_date_from_working_days(info_request.date_response_required_by, 15)
   end
 
-  module ClassMethods
-    def theme_display_status(status)
-      if status == 'deadline_extended'
-        _("Deadline extended.")
-      elsif status == 'wrong_response'
-        _("Wrong Response.")
-      else
-        raise _("unknown status ") + status
-      end
+  def display_status(status)
+    if status == 'deadline_extended'
+      _("Deadline extended.")
+    elsif status == 'wrong_response'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.